sky/clouds/azure.py

```diff
@@ -136,35 +136,39 @@ def get_default_instance_type(
     def _get_image_config(self, gen_version, instance_type):
         # az vm image list \
         #  --publisher microsoft-dsvm --all --output table
-        # nvidia-driver: 495.29.05, cuda: 11.5
-
-        # The latest image 2022.09.14/2022.08.11/22.06.10/22.05.11/
-        # 22.04.27/22.04.05 has even older nvidia driver 470.57.02,
-        # cuda: 11.4
+        # nvidia-driver: 535.54.03, cuda: 12.2
+        # see: https://github.com/Azure/azhpc-images/releases/tag/ubuntu-hpc-20230803
+        # All A100 instances is of gen2, so it will always use
+        # the latest ubuntu-hpc:2204 image.
         image_config = {
             'image_publisher': 'microsoft-dsvm',
-            'image_offer': 'ubuntu-2004',
-            'image_sku': '2004-gen2',
-            'image_version': '21.11.04'
+            'image_offer': 'ubuntu-hpc',
+            'image_sku': '2204',
+            'image_version': '22.04.2023080201'
         }
```

Comment:
Considering we are adding more different images for Azure, we should have the `image.csv` design for Asure as well. Can we add a TODO here for the refactoring in the future?